install.sh

```diff
@@ -1,234 +1,488 @@
+		fatal "[INSTALL] [CONFIGURE]" "Failed to retrieve the IP address of the gateway node"
+	info "[INSTALL] [CONFIGURE]" "Gateway node: ${GATEWAY} (${GATEWAY_IP})"
+}
+
+function install_liqo() {
+	info "[INSTALL]" "Installing Liqo on your cluster..."
+
+	# Ignore errors that may occur if the namespace already exists
+	${KUBECTL} create namespace "${LIQO_NAMESPACE}" 1>/dev/null 2>&1 || true
+
+	local LIQO_CHART="${TMPDIR}/${LIQO_CHARTS_PATH}"
+	${HELM} dependency update "${LIQO_CHART}" >/dev/null ||
+		fatal "[INSTALL]" "Something went wrong while installing Liqo"
+	${HELM} install liqo --kube-context "${KUBECONFIG_CONTEXT}" --namespace "${LIQO_NAMESPACE}" "${LIQO_CHART}" \
+		--set global.version="${LIQO_IMAGE_VERSION}" --set global.suffix="${LIQO_SUFFIX:-}" --set clusterName="${CLUSTER_NAME}" \
+		--set podCIDR="${POD_CIDR}" --set serviceCIDR="${SERVICE_CIDR}" --set gatewayIP="${GATEWAY_IP}" \
+		--set global.dashboard_ingress="${DASHBOARD_INGRESS:-}" >/dev/null ||
+			fatal "[INSTALL]" "Something went wrong while installing Liqo"
+
+	info "[INSTALL]" "Hooray! Liqo is now installed on your cluster"
+}
+
+function all_clusters_unjoined() {
+	local JSON_PATH="{.items[*].spec.join} {.items[*].status.incoming.joined} {.items[*].status.outgoing.joined}"
+	( ${KUBECTL} get foreignclusters --output jsonpath="${JSON_PATH}" 2>/dev/null || echo "" ) | \
+		grep --invert-match --silent "true"
+}
+
+function unjoin_clusters() {
+	# Do not fail in case of errors, to avoid exiting if Liqo had already been (partially) uninstaled
+	set +e
+
+	info "[UNINSTALL] [UNJOIN]" "Unjoining from all peers"
+
+	# Globally disable the broadcaster
+	local CLUSTER_CONFIG_PATCH='{"spec":{"advertisementConfig":{"outgoingConfig":{"enableBroadcaster":false}}}}'
+	${KUBECTL} patch clusterconfig configuration --patch "${CLUSTER_CONFIG_PATCH}" --type 'merge' >/dev/null 2>&1
+
+	# Avoid waiting it no peers were estabished
+	if all_clusters_unjoined; then
+		return 0
+	fi
+
+	# Set join=false to all ForeignCluster resources
+	FOREIGN_CLUSTERS=$(${KUBECTL} get foreignclusters --output jsonpath="{.items[*].metadata.name}") 2>/dev/null
+	for FOREIGN_CLUSTER in ${FOREIGN_CLUSTERS}; do
+		${KUBECTL} patch foreignclusters "${FOREIGN_CLUSTER}" --patch '{"spec":{"join":false}}' --type 'merge' >/dev/null 2>&1
+	done
+
+	info "[UNINSTALL] [UNJOIN]" "Waiting for the unjoining process to complete..."
+
+	local RETRIES=60
+	while ! all_clusters_unjoined; do
+		RETRIES=$(( RETRIES-1 ))
+		[ "${RETRIES}" -gt 0 ] ||
+			fatal "[UNINSTALL] [UNJOIN]" "Timeout: impossible to unpeer from all clusters"
+		sleep 1
+	done
+
+	info "[UNINSTALL] [UNJOIN]" "Waiting for the network operators to reconcile..."
+	# This sleep is definitely suboptimal, but it seems there is no alternative at the moment
+	sleep 10
+
+	set -e
+}
+
+function uninstall_liqo() {
+	# Do not fail in case of errors, to avoid exiting if Liqo had already been (partially) uninstaled
```

Comment:
small typo (**uninstaled**") here and in purge_liqo()

Reply:
Thanks!